charts/consul/templates/api-gateway-gatewayclass.yaml

```diff
@@ -0,0 +1,20 @@
+{{- if (or (and (ne (.Values.apiGateway.enabled | toString) "-") .Values.apiGateway.enabled) (and (eq (.Values.apiGateway.enabled | toString) "-") .Values.global.enabled)) }}
+{{- if .Values.apiGateway.managedGatewayClass.enabled }}
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: GatewayClass
+metadata:
+  name: consul-api-gateway
```

Comment:
This and GatewayClassConfig below would probably need to be prefixed by the `{{ template "consul.fullname" . }}` like we do for other resources

Reply:
Maybe the class config,  but I'm not sure that the class should. We want the gateway class thats provided by consul-k8s to be the same across installations so our documentation can have a deterministic name to reference. When a gateway is defined by an operator is must reference the gateway class which I don't think there should ever be more than one of for `consul-api-gateway`.

Reply:
Ah that makes a lot of sense!